lib/sink/s3.go

```diff
@@ -0,0 +1,68 @@
+package sink
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/MasteryConnect/skrape/lib/config"
+	"github.com/MasteryConnect/skrape/lib/mysqlutils"
+	"github.com/MasteryConnect/skrape/lib/skrape/skrapes3"
+	"github.com/apex/log"
+)
+
+type S3Sink struct {
+	*CsvSink
+	Cfg config.Config
+}
+
+func NewS3Sink(path, name string, bufferSize int, cfg config.Config) *S3Sink {
+	cs := &S3Sink{
+		Cfg:     cfg,
+		CsvSink: NewCsvSink(path, name, bufferSize),
+	}
+	return cs
+}
+
+func (s *S3Sink) ReadFinished() {
+	// Flush and close file
+	s.SinkCore.Close()
+	s.File.Close()
+	// Upload file
+	log.WithField("TableName", s.Name).Info("Uploading file")
+	skrapes3.Gzipload(s.Name, s.Path)
+	s.Schema()
+}
+
+func (s *S3Sink) Close() {
+	s.CsvSink.Close()
+}
+
+// Export a table schema to S3
+func (s *S3Sink) Schema() {
+	schema, paths := mysqlutils.TableSchema(s.Cfg.GetConn(), s.Name)
+
+	schemaname := s.Name + ".json"
+	pathsname := s.Name + "paths.json"
```

Comment:
Done.

On Tue, Aug 23, 2016 at 4:45 PM, Adam Haymond <notifications@github.com>
wrote:

> In lib/sink/s3.go
> <https://github.com/MasteryConnect/skrape/pull/3#discussion_r75965638>:
>
> > +	// Upload file
> > +	log.WithField("TableName", s.Name).Info("Uploading file")
> > +	skrapes3.Gzipload(s.Name, s.Path)
> > +	s.Schema()
> > +}
> > +
> > +func (s *S3Sink) Close() {
> > +	s.CsvSink.Close()
> > +}
> > +
> > +// Export a table schema to S3
> > +func (s *S3Sink) Schema() {
> > +	schema, paths := mysqlutils.TableSchema(s.Cfg.GetConn(), s.Name)
> > +
> > +	schemaname := s.Name + ".json"
> > +	pathsname := s.Name + "paths.json"
>
> will you change the string to read pathsname := s.Name + "_paths.json",
> it was an oversight that I corrected and apparently did not commit.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MasteryConnect/skrape/pull/3/files/fe30d7960618bd24b6856b3d9fb7db2850c27b80#r75965638>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFXQFgC1l-8H_y6nEmosJESpE6F5OMSJks5qi3gjgaJpZM4JrbH0>
> .
>